theano/gof/tests/test_wrapper.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import absolute_import, print_function, division
+import theano
+import numpy
+from unittest import TestCase
+from theano.gof import Op, COp, Apply
+from theano import Generic
+from theano.scalar import Scalar
+from theano.tensor import TensorType
+from theano.gof.wrapper import Wrapper, Wrap
+from theano import config
+from theano import tensor
+from theano.tests import unittest_tools as utt
+
+dtype = config.floatX
+tensor_type_0d = TensorType(dtype, tuple())
+scalar_type = Scalar(dtype)
+generic_type = Generic()
+
+
+# A test op to compute `y = a*x^2 + bx + c` for any tensor x, with a, b, c as op params.
+class QuadraticOpFunc(Op):
+    __props__ = ('a', 'b', 'c')
+    params_type = Wrapper(a=tensor_type_0d,
+                          b=scalar_type,
+                          c=generic_type)
```

Comment:
In fact, the test should probably be changed to always use 'float64' for a, b, c.  This would make more sense and be a better example.